code/modules/projectiles/guns/energy/special.dm

```diff
@@ -428,7 +429,14 @@
 	return ..()
 
 /obj/item/weapon/gun/energy/temperature/newshot()
-	..(temperature, e_cost)
+	if(!ammo_type || !power_supply)
+		return
+	var/obj/item/ammo_casing/energy/shot = ammo_type[select]
+	if(power_supply.charge >= shot.e_cost) //if there's enough power in the power_supply cell...
+		chambered = shot //...prepare a new shot based on the current ammo type selected
+		chambered.newshot(temperature, e_cost)
+	return

```

Comment:
This shouldn't change anything. It's just copy-pasted from [the parent proc](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/projectiles/guns/energy.dm#L83-L90).
That said, you might be able to completely remove `e_cost` from this `..()` call, assuming you change [these](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/projectiles/guns/energy/special.dm#L464-L477) lines to directly set `ammo_type[select]`'s `e_cost`, although you have to do some typecasting first.